### DISCUSSION TITLE
Can we make DataGridColumn.CurrentDirection property public?

### DISCUSSION THREAD

**Discussion by andregeuze** [Megabit/Blazorise#1707] [Ideas]
Hello!

I'm creating an external view of the Grid's sorted columns. The current implementation of the DataGridColumn however does not allow me to extract this information.

Therefore I would like to propose a small change; Is there a possibility to change the internal property CurrentDirection to public?
https://github.com/stsrki/Blazorise/blob/dd2f7a4cae0c51f913b979e4f7791320c2ab184e/Source/Extensions/Blazorise.DataGrid/DataGridColumn.cs#L138-L142

Proposed change:
```
public SortDirection CurrentDirection
{
    get => currentDirection[ParentDataGrid.SortMode];
    set => currentDirection[ParentDataGrid.SortMode] = value;
}
```

If you agree, please let me know. I can always implement it myself if you don't have the time 😎👍

**Comment by stsrki** [comment]
It's not that easy. If it's only the `getter` it shouldn't be a problem. Making it public would also require to implement `setter` logic. For example, if you would update `CurrentDirection` you would also expect `DataGrid` to update its state and refresh. Not to mention there should be `CurrentDirectionChanged` handler.

In fact, now that I think it more, the actual parameter that you need would be a `Direction`. But either case, without doing all of the things mentioned, it would just be half done job.

**Reply by andregeuze** [reply]
A public getter would help me. It's already possible to set the direction of a column by inheriting DataGrid in a custom class, followed by calling `OnSortClicked`.

**Reply by andregeuze** [reply]
@stsrki So how do you feel about making the getter public in a future version?
